test/uk/gov/hmrc/tai/viewModels/TaxAccountSummaryViewModelSpec.scala

```diff
@@ -441,46 +437,78 @@ class TaxAccountSummaryViewModelSpec extends PlaySpec with FakeTaiPlayApplicatio
+  val empEmployment2 = Employment("Employer name2", Some("1ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST2", "PAYE2", 2, None, false, false)
+  val pensionEmployment3 = Employment("Pension name1", Some("3ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST3", "PAYE3", 3, None, false, false)
+  val pensionEmployment4 = Employment("Pension name2", Some("4ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST4", "PAYE4", 4, None, false, false)
+  val empEmployment9 = Employment("Employer name3", Some("9ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST9", "PAYE9", 9, None, false, false)
+  val empEmployment10 = Employment("Employer name4", Some("10ABC"), new LocalDate(2017, 3, 1), Some(new LocalDate(2018, 4, 21)), Seq.empty[AnnualAccount], "DIST10", "PAYE10", 10, None, false, false)
+
+  val employments: Seq[Employment] = Seq(
+    empEmployment1,
+    empEmployment2,
+    pensionEmployment3,
+    pensionEmployment4,
     Employment("JobSeekerAllowance name1", Some("5ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST5", "PAYE5", 5, None, false, false),
     Employment("JobSeekerAllowance name2", Some("6ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST6", "PAYE6", 6, None, false, false),
     Employment("OtherIncome name1", Some("7ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST7", "PAYE7", 7, None, false, false),
     Employment("OtherIncome name2", Some("8ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST8", "PAYE8", 8, None, false, false),
-    Employment("Employer name3", Some("9ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST9", "PAYE9", 9, None, false, false),
-    Employment("Employer name4", Some("10ABC"), new LocalDate(2017, 3, 1), Some(new LocalDate(2018, 4, 21)), Seq.empty[AnnualAccount], "DIST10", "PAYE10", 10, None, false, false),
+    empEmployment9,
+    empEmployment10,
     Employment("Pension name3", Some("11ABC"), new LocalDate(2017, 3, 1), None, Seq.empty[AnnualAccount], "DIST11", "PAYE11", 11, None, false, false),
     Employment("Pension name4", Some("12ABC"), new LocalDate(2017, 3, 1), Some(new LocalDate(2018, 4, 21)), Seq.empty[AnnualAccount], "DIST12", "PAYE12", 12, None, false, false)
   )
 
+  val livePensionIncomeSources: Seq[IncomeSource] = Seq(
+    IncomeSource(livePension3, pensionEmployment3),
+    IncomeSource(livePension4, pensionEmployment4)
+  )
+
+  val liveEmploymentIncomeSources: Seq[IncomeSource] = Seq(
+    IncomeSource(liveEmployment1, empEmployment1),
+    IncomeSource(liveEmployment2, empEmployment2)
+  )
+
+  val ceasedEmploymentIncomeSources: Seq[IncomeSource] = Seq(
+    IncomeSource(potentiallyCeasedEmployment9, empEmployment9),
+    IncomeSource(ceasedEmployment10, empEmployment10)
+  )
+
```

Comment:
Seeing this bit of set up code in tests again, is it needed now you've moved the logic out to the service?

Reply:
Addressed by moving all shared test data into trait as above